File: ManagingCompanies/UpdatingACompany/UpdatingACompany.java
package ManagingCompanies.UpdatingACompany;

import com.mashape.unirest.http.*;
import com.mashape.unirest.http.exceptions.UnirestException;

public class UpdatingACompany {
    public static void main(String[] args) throws UnirestException {
        Unirest.setTimeouts(0, 0);
        HttpResponse<String> response = Unirest
                .post("https://api.apexchat.com/Services/ApexChatService.svc/companies/{companyID}")
                .header("apexchat-company", "CompanyName").header("apexchat-username", "Username")
                .header("apexchat-password", "Password").header("Content-Type", "application/json")
                .body("{\r\n    \"address\": \"123 Elm Street\",\r\n    \"zip\": \"12345\"\r\n}").asString();
        System.out.println(response.getBody());
    }
}

File: ManagingCompanies/UpdatingACompany/UpdatingACompany.cs
using System;
using RestSharp;

namespace UpdatingACompany
{
    class UpdatingACompany
    {
        static void Main(string[] args)
        {
            var client = new RestClient("https://api.apexchat.com/Services/ApexChatService.svc/companies/{companyID}");
            var request = new RestRequest(Method.POST);
            request.AddHeader("apexchat-company", "CompanyName");
            request.AddHeader("apexchat-username", "Username");
            request.AddHeader("apexchat-password", "Password");
            request.AddHeader("Content-Type", "application/json");
            request.AddParameter("application/json", "{\r\n    \"address\": \"123 Elm Street\",\r\n    \"zip\": \"12345\"\r\n}", ParameterType.RequestBody);
            IRestResponse response = client.Execute(request);
            Console.WriteLine(response.Content);
        }
    }
}
